pkg/azurefile/nodeserver.go

```diff
@@ -390,20 +410,35 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		mountFsType := cifs
 		if protocol == nfs {
 			mountFsType = nfs
+			if encryptInTransit {
+				mountFsType = aznfs
+			}
+		}
+		if mountFsType == aznfs && !d.enableAzurefileProxy {
+			return nil, status.Error(codes.InvalidArgument, "encryptInTransit is only available when azurefile-proxy is enabled")
 		}
+
 		if err := prepareStagePath(cifsMountPath, d.mounter); err != nil {
 			return nil, status.Errorf(codes.Internal, "prepare stage path failed for %s with error: %v", cifsMountPath, err)
 		}
-		execFunc := func() error {
-			return SMBMount(d.mounter, source, cifsMountPath, mountFsType, mountOptions, sensitiveMountOptions)
-		}
-		timeoutFunc := func() error { return fmt.Errorf("time out") }
-		if err := volumehelper.WaitUntilTimeout(90*time.Second, execFunc, timeoutFunc); err != nil {
-			var helpLinkMsg string
-			if d.appendMountErrorHelpLink {
-				helpLinkMsg = "\nPlease refer to http://aka.ms/filemounterror for possible causes and solutions for mount errors."
+		if mountFsType == aznfs {
+			klog.V(2).Infof("encryptInTransit is enabled, mount by azurefile-proxy")
+			if err := d.mountWithProxy(ctx, source, cifsMountPath, mountFsType, mountOptions, sensitiveMountOptions); err != nil {
```

Comment:
it's better using the same `volumehelper.WaitUntilTimeout` func for aznfs mount

Reply:
only need to change the `execFunc`?

Reply:
aznfs mount was done in azurefile-proxy, nfs mount was the same as before, code was not changed